app/models/link_graph/node.rb

```diff
@@ -68,4 +80,20 @@ def to_h
       links: children,
     }
   end
+
+  def might_have_own_links?
```

Comment:
Why the shift from a definite sounding attribute `has_own_links` to a less-than-certain method name?